packages/valory/skills/market_creation_manager_abci/behaviours.py

```diff
@@ -474,51 +506,36 @@ def _get_markets_with_funds(
         return markets_with_liq
 
     def _get_condition_preparation_events(
-        self,
-        condition_ids: List[bytes],
-        from_block: int,
+        self, condition_ids: List[str]
     ) -> Generator[None, None, Optional[List[Dict[str, Any]]]]:
         """Get condition preparation events."""
-        contract_api_response = yield from self.get_contract_api_response(
-            performative=ContractApiMessage.Performative.GET_STATE,
-            contract_address=self.params.conditional_tokens_contract,
-            contract_id=str(ConditionalTokensContract.contract_id),
-            contract_callable="get_condition_preparation_events",
-            condition_ids=condition_ids,
-            from_block=from_block,
-        )
-        if contract_api_response.performative != ContractApiMessage.Performative.STATE:
-            self.context.logger.error(
-                f"Failed to get condition preparation events: {contract_api_response}"
-            )
-            return None
-        condition_preparation_events = cast(
-            Optional[List[Dict[str, Any]]],
-            contract_api_response.state.body.get("data", []),
+        query = CONDITIONS_QUERY.substitute(conditions=json.dumps(condition_ids))
+        response = yield from self.get_http_response(
```

Comment:
We will have to add a retry mechanism here, but first let's test if this works or not